src/types/actions.ts

```diff
@@ -35,7 +35,7 @@ export type ActionResult = {
 };
 
 export type DispatchFunc = (action: Action, getState?: GetStateFunc | null) => Promise<ActionResult>;
-export type ActionFunc = (dispatch: DispatchFunc, getState: GetStateFunc) => Promise<ActionResult|ActionResult[]>;
+export type ActionFunc = (dispatch: DispatchFunc, getState: GetStateFunc) => Promise<ActionResult|ActionResult[]> | ActionResult;
```

Comment:
can you elaborate on this change please?

Reply:
The previous signature required that all thunk actions were async, so it didn't allow for synchronous thunk actions. For example, the following is not currently allowed:
```javascript
function doSomethingTwice(): ActionFunc {
    return (dispatch) => { // Adding async makes this work, but it's not needed since this isn't actually asynchronous
        dispatch({type: 'something'});
        dispatch({type: 'something'});
    };
}
```

Reply:
Gotcha. Thanks!